### PR TITLE
Add and expose enable_api and enable_introspection capture options

### PR DESCRIPTION
**Pull request by pierricgimmig**
Add option to toggle introspection and manual instrumentation from the capture settings UI.

Note that the api toggle works only from disabled to enabled, we can't go from enabled to 
disabled yet. This will come in a subsequent PR.

Test: toggle introspection, toggle api from disabled to enabled.

### CODE REVIEW

**Comment by ronaldfw** [comment]
It looks like the enable introspection option will also be there in non-dev mode? If yes, do we want to expose this setting to users?

File: src/CaptureClient/include/CaptureClient/CaptureClient.h [hunk 1]
```diff
@@ -78,7 +78,7 @@ class CaptureClient {
       const absl::flat_hash_map<uint64_t, orbit_client_protos::FunctionInfo>& selected_functions,
       const TracepointInfoSet& selected_tracepoints, double samples_per_second,
       orbit_grpc_protos::UnwindingMethod unwinding_method, bool collect_scheduling_info,
-      bool collect_thread_state, bool enable_introspection,
+      bool collect_thread_state, bool enable_introspection, bool enable_api,
```
Comment (dpallotti) [inline src/CaptureClient/include/CaptureClient/CaptureClient.h:81]: Careful that the order of `enable_api` and `enable_introspection` is different in the .cpp.

File: src/OrbitQt/CaptureOptionsDialog.ui [hunk 2]
```diff
@@ -58,6 +58,20 @@
...
+          </property>
+         </widget>
+        </item>
+        <item>
+         <widget class="QCheckBox" name="introspectionCheckBox">
+          <property name="text">
+           <string>Enable introspection</string>
+          </property>
+         </widget>
+        </item>
```
Comment (dpallotti) [inline src/OrbitQt/CaptureOptionsDialog.ui:74]: Basically what Ronald said. I think that we should hide this when not in `--devmode`, and also make sure that the when not in `--devmode` introspection is disabled no matter what the last setting for it was.

**Review by antonrohr — COMMENTED** [review]
I agree with Ronald that we should think about whether we want to expose this to users. 

I have one nit, basically about clean commit history.

Maybe there is a more expressive naming than just `enable_api`, it is very generic. I dont know if `enable_orbit_api` is necessary better, just something to think about. 

Besides these points, everything looks good

File: src/OrbitQt/orbitmainwindow.cpp [hunk 3]
```diff
@@ -924,6 +928,8 @@ void OrbitMainWindow::on_actionCaptureOptions_triggered() {
 
   orbit_qt::CaptureOptionsDialog dialog{this};
   dialog.SetCollectThreadStates(settings.value(kCollectThreadStatesSettingKey, false).toBool());
+  dialog.SetEnableApi(settings.value(kEnableApiSettingKey, false).toBool());
```
Comment (dpallotti) [inline src/OrbitQt/orbitmainwindow.cpp:931]: Do we want the default to be false? Wouldn't the user expect to get manual instrumentation by default if they introduced it in their code? Or is there a reason why it should be false by default?

Comment (pierricgimmig) [reply src/OrbitQt/orbitmainwindow.cpp:931]: Yes you're right, I changed it to default to true, and same for introspection to maintain the same behavior as before by default. Thanks.

File: src/OrbitQt/CaptureOptionsDialog.ui (hunk 2 above)
Comment (pierricgimmig) [reply src/OrbitQt/CaptureOptionsDialog.ui:74]: Yep, done and done, thanks.

File: src/CaptureClient/include/CaptureClient/CaptureClient.h (hunk 1 above)
Comment (pierricgimmig) [reply src/CaptureClient/include/CaptureClient/CaptureClient.h:81]: yikes, thanks.

**Comment by pierricgimmig** [comment]
> It looks like the enable introspection option will also be there in non-dev mode? If yes, do we want to expose this setting to users?

Yeah was on the fence about this, but I think we should maintain the previous behavior which is to keep it as a devmode-only option. I made the change, thanks for bringing it up.

**Comment by pierricgimmig** [comment]
> I have one nit, basically about clean commit history.

The commits will be squashed in a single commit.

**Comment by pierricgimmig** [comment]
> Maybe there is a more expressive naming than just `enable_api`

Yeah, I agree that it is pretty generic, but we have an "Api" project, and "ApiFunctions", etc. so the naming is consistent. I think that if the api evolves into separate functionalities, then we might want to consider exposing multiple toggles more appropriately named, for now I'll leave it as is.

**Review by antonrohr — APPROVED** [review]
LGTM, thank you for addressing the comments!

File: src/OrbitClientGgp/ClientGgp.cpp [hunk 4]
```diff
@@ -105,15 +105,17 @@ bool ClientGgp::RequestStartCapture(ThreadPool* thread_pool) {
...
   UnwindingMethod unwinding_method = options_.use_framepointer_unwinding
                                          ? UnwindingMethod::kFramePointerUnwinding
                                          : UnwindingMethod::kDwarfUnwinding;
 
   Future<ErrorMessageOr<CaptureOutcome>> result = capture_client_->Capture(
       thread_pool, target_process_->pid(), module_manager_, selected_functions_,
       selected_tracepoints, options_.samples_per_second, unwinding_method, collect_scheduling_info,
-      collect_thread_state, enable_introspection, max_local_marker_depth_per_command_buffer, false,
-      0, CaptureEventProcessor::CreateForCaptureListener(this, absl::flat_hash_set<uint64_t>{}));
+      collect_thread_state, enable_introspection, enable_api,
```
Comment (dpallotti) [inline src/OrbitClientGgp/ClientGgp.cpp:116]: `enable_api` should precede `enable_introspection` here.

**Review by dpallotti — APPROVED** [review]
Please fix the wrong order in `ClientGgp.cpp`. Looks good otherwise.

File: src/OrbitClientGgp/ClientGgp.cpp (hunk 4 above)
Comment (pierricgimmig) [reply src/OrbitClientGgp/ClientGgp.cpp]: Arrgh, sorry, properly fixed now, thanks.

**Pull request merged**